server/src/test/java/org/candlepin/pinsetter/tasks/PopulateHostedDBTaskTest.java

```diff
@@ -0,0 +1,187 @@
+        p1.addContent(TestUtil.createContent("p1-content-1"));
+        Product p2 = TestUtil.createProduct("prod2", "prod2");
+        p2.addContent(TestUtil.createContent("p2-content-1"));
+        p2.addContent(TestUtil.createContent("p2-content-2"));
+        p2.setDependentProductIds(dependentProductIds1);
+        Product p3 = TestUtil.createProduct("prod3", "prod3");
+        p3.addContent(TestUtil.createContent("p3-content-1"));
+        p3.addContent(TestUtil.createContent("p3-content-2"));
+        p3.addContent(TestUtil.createContent("p3-content-3"));
+
+        Product dp1 = TestUtil.createProduct("dprod1", "dprod1");
+        dp1.addContent(TestUtil.createContent("dp1-content-1"));
+        Product dp2 = TestUtil.createProduct("dprod2", "dprod2");
+        dp2.addContent(TestUtil.createContent("dp2-content-1"));
+        dp2.setDependentProductIds(dependentProductIds2);
+        Product dp3 = TestUtil.createProduct("dprod3", "dprod3");
+        dp3.addContent(TestUtil.createContent("dp3-content-1"));
+        dp3.addContent(TestUtil.createContent("dp3-content-2"));
+        Product dp4 = TestUtil.createProduct("dprod4", "dprod4");
+        dp4.addContent(TestUtil.createContent("dp4-content-1"));
+        dp4.addContent(TestUtil.createContent("dp4-content-2"));
+
+        HashMap<String, Product> productMap = new HashMap<String, Product>();
+        productMap.put(p1.getId(), p1);
+        productMap.put(p2.getId(), p2);
+        productMap.put(p3.getId(), p3);
+        productMap.put(dp1.getId(), dp1);
+        productMap.put(dp2.getId(), dp2);
+        productMap.put(dp3.getId(), dp3);
+        productMap.put(dp4.getId(), dp4);
+
+        LinkedList<Product> products = new LinkedList<Product>();
+        products.add(p1);
+        products.add(p2);
+        products.add(p3);
+
+        LinkedList<Product> dependentProducts1 = new LinkedList<Product>();
+        dependentProducts1.add(dp1);
+        dependentProducts1.add(dp2);
+        dependentProducts1.add(p3);
+
+        LinkedList<Product> dependentProducts1b = new LinkedList<Product>();
+        dependentProducts1b.add(dp1);
+        dependentProducts1b.add(dp2);
+
+        LinkedList<Product> dependentProducts2 = new LinkedList<Product>();
+        dependentProducts2.add(dp3);
+        dependentProducts2.add(dp4);
+        dependentProducts2.add(dp2);
+
+        LinkedList<Product> dependentProducts2b = new LinkedList<Product>();
+        dependentProducts2b.add(dp3);
+        dependentProducts2b.add(dp4);
+
+        when(psa.getProductsByIds(productIds)).thenReturn(products);
+        when(psa.getProductsByIds(dependentProductIds1)).thenReturn(dependentProducts1);
+        when(psa.getProductsByIds(dependentProductIds1b)).thenReturn(dependentProducts1b);
+        when(psa.getProductsByIds(dependentProductIds2)).thenReturn(dependentProducts2);
+        when(psa.getProductsByIds(dependentProductIds2b)).thenReturn(dependentProducts2b);
```

Comment:
I can't quite spot why there are 5 collections of product IDs, wouldn't it just be one call to the job, one call for all product IDs?